HelpSource/Guides/WritingClasses.schelp

```diff
@@ -161,54 +161,136 @@ MyClass : SomeSuperclass {
+Here are some more examples:
+code::
+Derivied(c: 10) // a: nil, b: nil, c: 10
+Derivied(1, 2, c: 10) // a: 1, b: 2, c: 10
+Derivied(d: 10) // If 'd' exisited somewhere in the heirachy, it would be assigned the value 10.
 
-Warning:: If the superclass also happened to call super.new.init it will have expected to call the .init method defined
-in that class (the superclass), but instead the message .init will find the implementation of the class that the object
-actually is, which is our new subclass. In such cases, a unique method name like myclassInit should be used.
 ::
 
-One easy way to copy the arguments passed to the instance variables when creating a class is to use link::Classes/Object#*newCopyArgs::.  This
-method will copy the arguments to the instance variables in the order that the variables were defined in the class,
-starting from the parent classes and working it's way down to the current class.
+Although this approach might seem complex, the benifit is that you can add new members to either class and it will always be clear which member you meant to assign to.
+If it is desirable to be able to remove instance members in the future,
+it is advised to require only keyword arguments be used by removing the named arguments and throwing an error if code::args:: is not empty.
+
+Another benifit is that the instance is never in an incomplete state, either all the instance members have been assigned to, or the object does not exist.
+
+The disadvantage come when you have complex initialisation logic where the arguments in code::Derived:: require knowledge of the arguments in code::Base::.
+Traditionally in SuperCollider this has been solved with an code::init:: method as seen below, although a 'builder' pattern might be considered for complex cases.
 
-code::
-MyClass {
-	var a,b,c;
 
-	*new { | a, b, c |
-		^super.newCopyArgs(a, b, c)
+code::
+Base {
+	var a, b;
+	*new {|a, b| ^super.new.init(a, b) }
+	init {|newA, newB|
+		// Initialsation logic goes here.
+		a = newA;
+		b = newB;
 	}
 }
+::
+
+This work but becomes more complex in inheritance chains.
 
-MyChildClass : MyClass {
-	var d;
+code::
+Base {
+	var a, b;
+	*new {|a, b| ^super.new.init(a, b) }
+	init {|newA, newB|
+		// Initialsation logic goes here.
+		a = newA;
+		b = newB;
+	}
+}
 
-	*new { | a, b, c, d |
-		^super.newCopyArgs(a, b, c, d)
+Derivied : Base {
+	var c;
+	*new { |a, b, c| ^super.new.deriviedInit(a, b, c) }
+	deriviedInit { |newA, newB, newC|
+		// do initialisation with all the arguments
+		this.init(newA, newB) // call Base.init
+		c = newC; // assign to Derivied's c
 	}
 }
 ::
 
+Warning:: If the superclass also happened to call super.new.init it will have expected to call the .init method defined
+in that class (the superclass), but instead the message .init will find the implementation of the class that the object
+actually is, which is our new subclass.
+In such cases, a unique method name like myclassInit should be used.
```

Comment:
or better, a specialised name that indicates what kind of initialization it performs.